syntax="proto3";

message PlayerState{
	int32 hp=1;
	float move_dir=2;
	float aim_dir=3;
	bool is_moving=4;
	float shoot_cd_left=5;
	repeated float terrain_info=6;
}

message RegionInfo {
	int32 amount=1;
	int32 dir_info=2;
}

message Action {
	float move_dir = 1;
	float aim_dir = 2;
	bool move_state = 3;
	bool shoot_state = 4;
}

message SensorData{
	PlayerState player_state = 1;
	repeated RegionInfo mob_data =2;
	repeated RegionInfo mob_bullet_data = 3;
	repeated RegionInfo player_bullet_data =4;
}

message ServerMsg {
	int32 field_id = 1;
	int32 id = 2;
	int32 msg_type = 3;
	SensorData sensor_data=4;
	float reward=5;
	bool game_end=6;
}

message ClientMsg{
	int32 field_id = 1;
	int32 id = 2;
	int32 msg_type = 3;
	Action action = 4;
}

message ServerCtrlMsg {
	int32 cmd = 1;
}